source/common/rate_limiter_impl.h

```diff
@@ -203,4 +204,30 @@ class GraduallyOpeningRateLimiterFilter : public FilteringRateLimiterImpl {
   const std::chrono::nanoseconds ramp_time_;
 };
 
+/**
+ * Thin wrapper around absl::zipf_distribution that will pull zeroes and ones from the distribution
+ * with the intent to probabilistically suppress the wrapped rate limiter.
+ * This may need further consideration, because it will shoot holes in the pacing, lowering the
+ * actual achieved frequency.
+ */
+class ZipfRateLimiterImpl : public FilteringRateLimiterImpl {
+public:
+  /**
+   * From the absl header associated to the zipf distribution:
+   * Preconditions: v > 0, q > 1
+   * The precondidtions are validated when NDEBUG is not defined via
+   * a pair of assert() directives.
+   * If NDEBUG is defined and either or both of these parameters take invalid
+   * values, the behavior of the class is undefined.
+   */
+  ZipfRateLimiterImpl(RateLimiterPtr&& rate_limiter, bool deterministic, double q = 2.0,
```

Comment:
I believe we had some open comments for the API and its documentation here in the original PR where this was introduced before we split it out. Would you mind resolving those comments before we proceed?

Reply:
Done, addressed the [prior feedback round](https://github.com/envoyproxy/nighthawk/pull/218#pullrequestreview-335035033) in 10dbc29f4affec32379b40a4e0d9a3f117a36474